src/main/scala/cromwell/engine/backend/sge/SgeBackend.scala

```diff
@@ -0,0 +1,126 @@
+import cromwell.parser.BackendType
+import cromwell.util.FileUtil._
+import scala.concurrent.{ExecutionContext, Future}
+
+import scala.annotation.tailrec
+import scala.language.postfixOps
+import scala.sys.process._
+import scala.util.{Failure, Success, Try}
+
+class SgeBackend extends Backend with LocalFileSystemOperations with LazyLogging {
+  import LocalBackend.WriteWithNewline
+
+  override def bindCall(workflowDescriptor: WorkflowDescriptor,
+                        call: Call,
+                        locallyQualifiedInputs: CallInputs,
+                        abortRegistrationFunction: AbortFunctionRegistration): BackendCall = {
+    SgeBackendCall(this, workflowDescriptor, call, locallyQualifiedInputs, abortRegistrationFunction)
+  }
+
+  override def execute(backendCall: BackendCall): Try[CallOutputs] = backendCall match {
+    case bc: SgeBackendCall => execute(bc)
+    case _ => Failure(new UnsupportedOperationException("SgeBackend can only execute SgeBackendCalls"))
+  }
+
+  /**
+   * Executes the specified command line, using the supplied lookup function for expression evaluation.
+   * Returns a `Map[CallOutputs]` of output names to values.
+   */
+  private def execute(backendCall: SgeBackendCall): Try[CallOutputs]  = {
+    val tag: String = s"${this.getClass.getName} [UUID(${backendCall.workflowDescriptor.shortId}):${backendCall.call.name}]"
+
+    backendCall.instantiateCommand match {
+      case Success(instantiatedCommand) =>
+        logger.info(s"$tag `$instantiatedCommand`")
+
+        val (scriptFile, scriptWriter) = backendCall.script.fileAndWriter
+        scriptWriter.writeWithNewline("#!/bin/sh")
+        scriptWriter.writeWithNewline(instantiatedCommand)
+        scriptWriter.writeWithNewline("echo $? > rc")
+        scriptWriter.flushAndClose()
+
+        val sgeJobName = s"cromwell_${backendCall.workflowDescriptor.shortId}_${backendCall.call.name}"
+        val argv = Seq("qsub", "-N", sgeJobName, "-V", "-b", "n", "-wd", backendCall.callRootPath.toAbsolutePath, "-o", backendCall.stdout.getFileName, "-e", backendCall.stderr.getFileName, backendCall.script.toAbsolutePath).map(_.toString)
+        val backendCommandString = argv.map(s => "\""+s+"\"").mkString(" ")
+        logger.info(s"$tag backend command: $backendCommandString")
+
+        val (qsubStdoutFile, qsubStdoutWriter) = backendCall.callRootPath.resolve("qsub.stdout").fileAndWriter
+        val (qsubStderrFile, qsubStderrWriter) = backendCall.callRootPath.resolve("qsub.stderr").fileAndWriter
+        val process = argv.run(ProcessLogger(qsubStdoutWriter writeWithNewline, qsubStderrWriter writeWithNewline))
+        val rc: Int = process.exitValue()
+        Vector(qsubStdoutWriter, qsubStderrWriter) foreach { _.flushAndClose() }
+        logger.info(s"$tag qsub rc=$rc")
+
+        "^Your job (\\d+)".r.findFirstMatchIn(qsubStdoutFile.slurp) match {
+          case Some(m) =>
+            val sgeJobId = m.group(1).toInt
```

Comment:
I realize this is WIP so ignore if this is just temporary but I'll submit that a 60 line case block containing multiple sub functions could be rejiggered